extensions/amp-form/0.1/amp-form.js

```diff
@@ -88,11 +92,14 @@ export class AmpForm {
         method: this.method_,
         credentials: 'include',
         requireAmpResponseSourceOrigin: true,
-      }).then(() => this.setState_(FormState_.SUBMIT_SUCCESS))
-          .catch(error => {
-            this.setState_(FormState_.SUBMIT_ERROR);
-            rethrowAsync('Form submission failed:', error);
-          });
+      }).then(response => {
+        this.setState_(FormState_.SUBMIT_SUCCESS);
+        this.renderTemplate_(FormState_.SUBMIT_SUCCESS, response);
+      }).catch(error => {
+        this.setState_(FormState_.SUBMIT_ERROR);
+        this.renderTemplate_(FormState_.SUBMIT_ERROR, error).then(
+            rethrowAsync.bind(null, 'Form submission failed:', error));
```

Comment:
How do we avoid this? I might need to re-write the way I am testing this. Will try to do some more tests here.